tests/test_meta/test_grouped_transformer.py

```diff
@@ -0,0 +1,326 @@
+
+    # Top row should all be equal to the small value of the range, bottom the other
+    assert np.allclose(transformed[0, :], scaling_range[0])
+    assert np.allclose(transformed[1, :], scaling_range[1])
+
+
+def test_missing_groups_transform_noglobal(dataset_with_single_grouping, scaling_range):
+    X, y, groups, X_with_groups, grouper = dataset_with_single_grouping
+
+    trf = MinMaxScaler(scaling_range)
+    transformer = GroupedTransformer(trf, groups=grouper, use_global_model=False)
+    transformer.fit(X_with_groups, y)
+
+    # Array with 2 rows, first column a new group. Remaining top are out of range so should be the range
+    X_test = np.concatenate([
+        np.array([[3], [3]]), np.stack([X.min(axis=0) - 1, X.max(axis=0) + 1], axis=0)
+    ], axis=1)
+
+    with pytest.raises(ValueError):
+        transformer.transform(X_test)
+
+
+@pytest.fixture()
+def multiple_obs_fitter():
+    from sklearn.base import BaseEstimator, TransformerMixin
+
+    class MultipleObsFitter(BaseEstimator, TransformerMixin):
+        """A transformer that needs more than 1 value to fit"""
+        def fit(self, X, y=None):
+            if len(X) <= 1:
+                raise ValueError("Need more than 1 value to fit")
+
+            return self
+
+        def transform(X):
+            return X
+
+    return MultipleObsFitter()
+
+
+def test_exception_in_group(multiple_obs_fitter):
+    X = np.array([
+        [1, 2],
+        [1, 0],
+        [2, 1],
+    ])
+
+    # Only works on groups greater than 1, so will raise an error in group 2
+    transformer = GroupedTransformer(multiple_obs_fitter, groups=0, use_global_model=False)
+
+    with pytest.raises(ValueError) as e:
+        transformer.fit(X)
+
+        assert "group 2" in str(e)
+
+
+def test_array_with_strings():
+    X = np.array([
+        ("group0", 2),
+        ("group0", 0),
+        ("group1", 1),
+        ("group1", 3),
+    ], dtype='object')
+
+    trf = MinMaxScaler()
+    transformer = GroupedTransformer(trf, groups=0, use_global_model=False)
+    transformer.fit_transform(X)
+
+
+@pytest.fixture(scope="module")
+def penguins_df():
+    df = load_penguins(as_frame=True).dropna()
+    X = df.drop(columns='species')
+
+    return X
+
+
+@pytest.fixture(scope="module")
+def penguins(penguins_df):
+    return penguins_df.values
+
+
+def test_df(penguins_df):
+    meta = GroupedTransformer(StandardScaler(), groups=["island", "sex"])
+
+    # This should work fine
+    meta.fit_transform(penguins_df)
+
+
+def test_df_missing_group(penguins_df):
+    meta = GroupedTransformer(StandardScaler(), groups=["island", "sex"])
+
+    # Otherwise the fixture is changed
+    X = penguins_df.copy()
+    X.loc[0, "island"] = None
+
+    with pytest.raises(ValueError):
+        meta.fit_transform(X)
+
+
+def test_array_with_multiple_string_cols(penguins):
+    X = penguins
+
+    meta = GroupedTransformer(StandardScaler(), groups=[0, -1])
+
+    # This should work fine
+    meta.fit_transform(X)
```

Comment:
Let's add an `assert` here. I've noticed that we still haven't checked for the output shape. Would be good to do here.